src/views/checkups/network/hooks/useCheckupsNetworkPermissions.ts

```diff
@@ -0,0 +1,82 @@
+import { useMemo } from 'react';
+
+import {
+  ClusterRoleBindingModel,
+  ClusterRoleModel,
+  modelToGroupVersionKind,
+  ServiceAccountModel,
+} from '@kubevirt-ui/kubevirt-api/console';
+import {
+  IoK8sApiCoreV1ServiceAccount,
+  IoK8sApiRbacV1ClusterRole,
+  IoK8sApiRbacV1ClusterRoleBinding,
+} from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { useActiveNamespace, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+import {
+  findObjectByName,
+  KIAGNOSE_CONFIGMAP_ACCESS,
+  KUBEVIRT_VM_LATENCY_CHECKER,
+  VM_LATENCY_CHECKUP_SA,
+} from '../utils/utils';
+
+const useCheckupsNetworkPermissions = (): { isPermitted: boolean; loading: boolean } => {
+  const [namespace] = useActiveNamespace();
+
+  const [serviceAccounts, loadingServiceAccounts] = useK8sWatchResource<
+    IoK8sApiCoreV1ServiceAccount[]
+  >({
+    groupVersionKind: modelToGroupVersionKind(ServiceAccountModel),
+    isList: true,
+    namespace,
+  });
+
+  const [roles, loadingRoles] = useK8sWatchResource<IoK8sApiRbacV1ClusterRole[]>({
+    groupVersionKind: modelToGroupVersionKind(ClusterRoleModel),
+    isList: true,
+  });
+
+  const [roleBinding, loadingRolesBinding] = useK8sWatchResource<
+    IoK8sApiRbacV1ClusterRoleBinding[]
+  >({
+    groupVersionKind: modelToGroupVersionKind(ClusterRoleBindingModel),
+    isList: true,
+  });
+  const isLatencyRole = useMemo(
+    () => findObjectByName(roles, KUBEVIRT_VM_LATENCY_CHECKER),
+    [roles],
+  );
+
+  const isConfigMapRole = useMemo(
+    () => findObjectByName(roles, KIAGNOSE_CONFIGMAP_ACCESS),
+    [roles],
+  );
+
+  const isLatencyRoleBinding = useMemo(
+    () => findObjectByName(roleBinding, KUBEVIRT_VM_LATENCY_CHECKER),
+    [roleBinding],
+  );
+
+  const isConfigMapRoleBinding = useMemo(
+    () => findObjectByName(roleBinding, KIAGNOSE_CONFIGMAP_ACCESS),
+    [roleBinding],
+  );
+
+  const isServiceAccount = useMemo(
+    () => findObjectByName(serviceAccounts, VM_LATENCY_CHECKUP_SA),
+    [serviceAccounts],
+  );
```

Comment:
Do we know the namespace of those resources? `default` or maybe the current `namespace`?
In that case, we could watch directly the resource and see if the error code is 404

Reply:
its the active/current namespace, for serviceAccounts , the first call, as I used there, the other resources I did tried using the namespace but no luck :( 
I can try again to be sure